src/fancy-tools/scaleOverlay.js

```diff
@@ -0,0 +1,242 @@
+  });
+};
+
+/**
+ * @param  {} context
+ * @param  {} imageAttributes
+ */
+const drawVerticalScalebarIntervals = (context, imageAttributes) => {
+  let i = 0;
+
+  while (imageAttributes.verticalLine.start.y + i * imageAttributes.verticalMinorTick <= imageAttributes.vscaleBounds.bottomRight.y) {
+
+    const startPoint = {
+      x: imageAttributes.verticalLine.start.x,
+      y: imageAttributes.verticalLine.start.y + i * imageAttributes.verticalMinorTick
+    };
+
+    const endPoint = {
+      x: 0,
+      y: imageAttributes.verticalLine.start.y + i * imageAttributes.verticalMinorTick
+    };
+
+    if (i % 5 === 0) {
+
+      endPoint.x = imageAttributes.verticalLine.start.x - imageAttributes.majorTickLength;
+    } else {
+
+      endPoint.x = imageAttributes.verticalLine.start.x - imageAttributes.minorTickLength;
+    }
+
+    drawLine(context, startPoint, endPoint);
+
+    i++;
+  }
+};
+
+const drawHorizontalScalebarIntervals = (context, imageAttributes) => {
+  let i = 0;
+
+  while (imageAttributes.horizontalLine.start.x + i * imageAttributes.horizontalMinorTick <= imageAttributes.hscaleBounds.bottomRight.x) {
+
+    const startPoint = {
+      x: imageAttributes.horizontalLine.start.x + i * imageAttributes.horizontalMinorTick,
+      y: imageAttributes.horizontalLine.start.y
+    };
+
+    const endPoint = {
+      x: imageAttributes.horizontalLine.start.x + i * imageAttributes.horizontalMinorTick,
+      y: 0
+    };
+
+    if (i % 5 === 0) {
+      endPoint.y = imageAttributes.horizontalLine.start.y - imageAttributes.majorTickLength;
+    } else {
+      endPoint.y = imageAttributes.horizontalLine.start.y - imageAttributes.minorTickLength;
+    }
+
+    drawLine(context, startPoint, endPoint);
+
+    i++;
+  }
+};
+
+function drawVerticalScalebar (context, imageAttributes) {
+  const startPoint = {
+    x: imageAttributes.verticalLine.start.x,
+    y: imageAttributes.verticalLine.start.y
+  };
+  const endPoint = {
+    x: imageAttributes.verticalLine.end.x,
+    y: imageAttributes.verticalLine.end.y
+  };
+
+  const { color, lineWidth } = imageAttributes;
+
+  path(context, { color,
+    lineWidth }, (context) => {
+    drawLine(context, startPoint, endPoint);
+    drawVerticalScalebarIntervals(context, imageAttributes);
+  });
+}
+
+function drawHorizontalScalebar (context, imageAttributes) {
+  const startPoint = {
+    x: imageAttributes.horizontalLine.start.x,
+    y: imageAttributes.horizontalLine.start.y
+  };
+  const endPoint = {
+    x: imageAttributes.horizontalLine.end.x,
+    y: imageAttributes.horizontalLine.end.y
+  };
+
+  drawLine(context, startPoint, endPoint);
+  drawHorizontalScalebarIntervals(context, imageAttributes);
+}
+
+export default class extends baseTool {
+  constructor (name) {
+    super({
+      name: name || 'scaleOverlay',
+      configuration: {
+        color: 'white',
+        lineWidth: 2,
+        shadowColor: 'black',
+        shadowBlur: 4,
```

Comment:
We might want to keep track of what common properties we may want to have a global setting/default for, and then allow a local value to override. That way it's easier to set values that affect all tools, and override them as needed.

Reply:
You mean I should be using the `stateManagement/` to get those values?